infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/configurator/SshKeysConfigurator.java

```diff
@@ -0,0 +1,209 @@
+import static java.util.stream.Collectors.toList;
+import static org.eclipse.che.workspace.infrastructure.kubernetes.Constants.DEV_WORKSPACE_MOUNT_LABEL;
+import static org.eclipse.che.workspace.infrastructure.kubernetes.Constants.DEV_WORKSPACE_MOUNT_PATH_ANNOTATION;
+import static org.eclipse.che.workspace.infrastructure.kubernetes.Constants.DEV_WORKSPACE_WATCH_SECRET_LABEL;
+import static org.eclipse.che.workspace.infrastructure.kubernetes.namespace.KubernetesObjectUtil.isValidConfigMapKeyName;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.api.model.SecretBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import jakarta.validation.constraints.NotNull;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+import javax.inject.Inject;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.ssh.server.SshManager;
+import org.eclipse.che.api.ssh.server.model.impl.SshPairImpl;
+import org.eclipse.che.api.ssh.shared.model.SshPair;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.api.workspace.server.spi.NamespaceResolutionContext;
+import org.eclipse.che.workspace.infrastructure.kubernetes.KubernetesClientFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class mounts existing user SSH Keys into a special Kubernetes Secret on user-s namespace.
+ */
+public class SshKeysConfigurator implements NamespaceConfigurator {
+
+  public static final String SSH_KEY_SECRET_NAME = "che-git-ssh-key";
+  private static final String SSH_KEYS_WILL_NOT_BE_MOUNTED_MESSAGE =
+      "Ssh keys %s have invalid names and can't be mounted to namespace %s.";
+
+  private static final String SSH_BASE_CONFIG_PATH = "/.ssh/";
+  private static final String SSH_PRIVATE_KEYS = "private";
+  private static final String SSH_PRIVATE_KEYS_PATH = SSH_BASE_CONFIG_PATH + SSH_PRIVATE_KEYS;
+
+  private final SshManager sshManager;
+
+  private final KubernetesClientFactory clientFactory;
+
+  private static final Logger LOG = LoggerFactory.getLogger(SshKeysConfigurator.class);
+
+  public static final Pattern VALID_DOMAIN_PATTERN =
+      Pattern.compile("^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$");
+
+  @Inject
+  public SshKeysConfigurator(SshManager sshManager, KubernetesClientFactory clientFactory) {
+    this.sshManager = sshManager;
+    this.clientFactory = clientFactory;
+  }
+
+  @Override
+  public void configure(NamespaceResolutionContext namespaceResolutionContext, String namespaceName)
+      throws InfrastructureException {
+
+    var client = clientFactory.create();
+    List<SshPairImpl> vcsSshPairs = getVcsSshPairs(namespaceResolutionContext);
```

Comment:
@amisevsk wondering if we need to process the internal ssh keys for async-storage. I'm not sure how the DevWorkspace handles the key management for it

Reply:
https://github.com/eclipse-che/che-server/blob/98b66ff1f64ebf2174b51e1a348d482590962e82/infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/AsyncStorageProvisioner.java#L204